rust/lance/src/dataset/fragment.rs

```diff
@@ -146,9 +146,13 @@ impl FileFragment {
     /// Open all the data files as part of the projection schema.
```

Comment:
I know this is unrelated to the PR but can we update this description?  I'm not sure I understand this sentence.

Maybe

```
Opens a FileFragment with a given default projection

All read operations (other than `read_projected`) will use the supplied
default projection.
```

Does `read_projected` give the intersection of the two projections?  Does it require the per-call projection to be narrower than the fragment open projection?

Reply:
Done.